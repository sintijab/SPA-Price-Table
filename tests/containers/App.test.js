import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store'
import Provider from 'react-redux';
import App from '../../js/containers/View';

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore();

describe('<App />', () => {
  let wrapper, store;

  beforeEach(() => {
      const initialState = {
        modalState: [
          {
            id: 1,
            name: "MartinFirma",
            budget: 10000,
            budget_spent: 45000,
            date_of_first_purchase: "2119-07-07"
          },
        ]
      };
      store = mockStore(initialState);
      wrapper = shallow(
          <App store={store} />
      );
  });

    it('should render App container', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
});
