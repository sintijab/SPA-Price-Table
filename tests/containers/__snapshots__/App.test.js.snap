// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App container 1`] = `
<ContextProvider
  value={null}
>
  <App
    dataEmpty={[Function]}
    dataFailed={[Function]}
    dataLoaded={[Function]}
    modalState={Array []}
    showModal={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
